components/InfiniteScrollCourses.js

```diff
@@ -0,0 +1,57 @@
+import { React, useState } from 'react'
+import PropTypes from 'prop-types'
+import Grid from '@mui/material/Grid'
+import CourseCard from '/components/courseCard'
+import { courseShape } from '/lib/prop-types'
+import InfiniteScroll from 'react-infinite-scroll-component'
+
+export default function InfiniteScrollCourses({ courses }) {
+  const perPage = 100
+  const [coursesSlice, setCourses] = useState(courses.slice(0, perPage))
+  const [scrollPosition, setScrollPosition] = useState(perPage)
+  const [hasMore, setHasMore] = useState(true)
+
+  const fetchData = () => {
+    const increaseTotalSlice = scrollPosition + perPage
+    if (increaseTotalSlice >= courses.length) {
+      setHasMore(false)
+    } else {
+      setScrollPosition(increaseTotalSlice)
+    }
+    setCourses(() => courses.slice(0, increaseTotalSlice))
+  }
+
+  const totalCoursesLessThanPerPage = perPage >= courses.length ? true : false
+
+  const CoursesGrid = () => (
+    <Grid container spacing={2}>
+      {coursesSlice.map((course) => (
+        <Grid key={course.course_number} item xs={12} sm={6}>
+          <CourseCard course={course} />
+        </Grid>
+      ))}
+    </Grid>
+  )
+
+  const Scroller = () => (
+    <InfiniteScroll
+      dataLength={coursesSlice.length}
+      next={fetchData}
+      hasMore={hasMore}
+      loader={<h4>Loading...</h4>}
+      endMessage={
+        <p style={{ textAlign: 'center' }}>
+          <b>Yay! You have seen it all</b>
```

Comment:
Could we replace this with a "Scroll to Top" link?
Something like this: 
![scroll to top](https://user-images.githubusercontent.com/33558504/219076792-d238bdb6-1212-4300-8136-20cf13a15789.png)